spec/README.md

```diff
@@ -53,6 +53,30 @@ when opportunities arrive.
 The `factory_bot` gem is used to create test data. Factories are defined in the
 `spec/factories` folder.
 
+`create_entire_hierarchy` is a helper method that creates a complete data
+hierarchy, with all necessary associations (e.g. an audio recording that
+belongs to a site that belongs to a region that belongs to a project).
+
+Factory cascading creation may result in unneccesary branches which could slow
+down tests. This can be avoided by specifying a value for the assocation upon
+creation.
+
+E.g. The items that will be created using the `audio_recording` factory with no
+arguments:
+
+                            audio_recording
+                            /             \
+                          site             user
+                       /    |    \
+                      /     |      user
+                region    project
+               /      \          \
+              user     \          user
+                     project
+                         \
+                        user
+
```

Comment:
this is great, but take advantage of GH built in support for mermaid diagrams. I reckon and ERD is the way to go: https://mermaid.js.org/syntax/entityRelationshipDiagram.html

Also do a compare and contrast of `create_entire_hierarchy` vs `create` with the diagrams (add a second diagram)